src/main/java/org/jboss/netty/handler/codec/http/HttpMessageDecoder.java

```diff
@@ -753,16 +753,55 @@ private String[] splitHeader(String sb) {
         };
     }
 
-    private static int findNonWhitespace(String sb, int offset) {
-        int result;
-        for (result = offset; result < sb.length(); result ++) {
-            if (!Character.isWhitespace(sb.charAt(result))) {
-                break;
+    private static int findNonSPLenient(String sb, int offset) {
+        for (int result = offset; result < sb.length(); ++result) {
+            char c = sb.charAt(result);
+            // See https://tools.ietf.org/html/rfc7230#section-3.5
+            if (isSPLenient(c)) {
+                continue;
+            }
+            if (Character.isWhitespace(c)) {
+                // Any other whitespace delimiter is invalid
+                throw new IllegalArgumentException("Invalid separator");
             }
+            return result;
         }
-        return result;
+        return sb.length();
+    }
+
+     private static int findSPLenient(String sb, int offset) {
+         for (int result = offset; result < sb.length(); ++result) {
+             if (isSPLenient(sb.charAt(result))) {
+                 return result;
+             }
+         }
+         return sb.length();
+     }
+ 
+     private static boolean isSPLenient(char c) {
+         // See https://tools.ietf.org/html/rfc7230#section-3.5
+         return c == ' ' || c == (char) 0x09 || c == (char) 0x0B || c == (char) 0x0C || c == (char) 0x0D;
+     }
+
+    private static int findNonWhitespace(String sb, int offset, boolean validateOWS) {
+        for (int result = offset; result < sb.length(); ++result) {
+            char c = sb.charAt(result);
+            if (!Character.isWhitespace(c)) {
+                return result;
+            } else if (validateOWS && !isOWS(c)) {
```

Comment:
Shouldn't it be just  `else if (!isOWS(c))` as in the referenced commit?

Reply:
````
Motivation:

We should validate that only OWS is allowed before / after a header name and otherwise throw. At the moment we just "strip" everything except OWS.
````

Reply:
Done.